### DISCUSSION TITLE
Riparazioni

### DISCUSSION THREAD

**Discussion by gideonklein** [indomus/forum#2523] [HUB: Home Assistant]
Qualcuno può aiutarmi? 
Mi trovo 2 riparazioni da fare in HA ma non riesco:
- La piattaforma dnsip per l'integrazione sensor non supporta l'impostazione della piattaforma.
- La configurazione Command Line sensor utilizzando YAML è stata spostata.

Grazie in anticipo

**Comment by snakuzzo** [comment]
Leggi tutte le breaking change quando fai gli aggiornamenti...sul sito ufficiale e/o sul nostro sito...

https://indomus.it/news/hot-aggiornamento-2023-6-per-home-assistant-core/

https://www.home-assistant.io/blog/2023/06/07/release-20236/

Di seguito la doc ufficiale delle tue integrazioni 

https://www.home-assistant.io/integrations/dnsip/


https://www.home-assistant.io/integrations/command_line/


Leggi bene e vedrai che risolverai

**Reply by gideonklein** [reply]
Ciao snakuzzo, grazie. Ho letto ma non riesco a risolvere, potete aiutarmi?
IP DNS. Ho trovato l'integrazione su home assistant già installata ma non risolve il problema
Per quanto riguarda i sensori non riesco a capire cosa devo mettere su command:

Ade esempio questo è un sensore che devo correggere:
  - platform: template
    sensors:
      energy_total:
        friendly_name: 'Total Energy'
        value_template: "{{ (states('sensor.sonoff_1000e10004_energy')|float + states('sensor.sonoff_1001121e93_energy')|float + states('sensor.sonoff_1000dd2821_energy')|float + states('sensor.sonoff_1000e7dbdd_energy')|float)|round(3) }}"
        unit_of_measurement: "kWh"

**Reply by snakuzzo** [reply]
Innanzi tutto devi usare il markdown quando utilizzi il codice.

Per quanto riguarda DNS IP; come è scritto in documentazione (che ti ho linkato), l'integrazione va fatta via webgui e non via YAML. Eventuale codice YAML relativo a DNS IP che avevi scritto lo devi cancellare (e riavviare Core ovviamente).

![image](https://github.com/indomus/forum/assets/7205371/dc35ede2-680c-4ab0-a574-2064e47f41a4)


Quanto a command line l'abbiamo scritto in maniera che più precisa non si può...

![image](https://github.com/indomus/forum/assets/7205371/2929c3d3-0819-42b2-aa4f-1c496d5abe69)

...se continui a non capire come fare, posta quello che hai e lo si corregge assieme.

Infine, il template sensor che hai postato non c'entra nulla con tutto il resto e, se è sbagliato, il problema è un altro e dovresti postare l'errore che hai.

**Comment by gideonklein** [comment]
Caro Diego, grazie per il tempo che perdete con me.
Purtroppo non riesco a risolvere, se mi aiutate passo passo magari riesco.
Allora DNS IP io non so nemmeno cos'è, non ho mai scritto una riga con questo su Yaml ma me lo sono trovato configurato come integrazione. Ho cercato nel mio home assistant è risulta solo una stringa trovata nel file di log "home-assistant.log.1" : 2023-07-02 19:07:13.691 ERROR (MainThread) [homeassistant.components.sensor] The dnsip platform for the sensor integration does not support platform setup. Please remove it from your config.

Per quanto riguarda i sensor non so come settare il command: cosa devo mettere?
Il codice che ho inviato ieri e che rimetto in maniera corretta con markdown è perfettamente funzionante la non tiene conto del command_line: quindi, se non lo correggo, non funzionerà piu' ad agosto "Questo smette di funzionare nella versione 2023.8.0. Risolvi il problema prima di effettuare l'aggiornamento"

```  - platform: template
    sensors:
      energy_total:
        friendly_name: 'Total Energy'
        value_template: "{{ (states('sensor.sonoff_1000e10004_energy')|float + states('sensor.sonoff_1001121e93_energy')|float + states('sensor.sonoff_1000dd2821_energy')|float + states('sensor.sonoff_1000e7dbdd_energy')|float)|round(3) }}"
        unit_of_measurement: "kWh" ```

**Reply by snakuzzo** [reply]
Per DNS IP, cerca nel codice dove hai scritto `dnsip` e rimuovi quel sensore. In qualche punto deve pur essere, altrimenti non ti darebbe errore. E su questo non ti possiamo aiutare noi, devi cercare tu dove sta...noi che ne possiamo sapere ?!

Quanto a command line, il codice che hai postato non c'entra una mazza con entità di tipo command line, ma è un un sensor template. Se ti dà problemi con entità di tipo command line, vale il discorso di cui sopra...cerca in tutti i tuoi file dove hai usato `command_line` e poi ti attrezzi per correggere

**Reply by gideonklein** [reply]
Grazie! Ho capito e ho risolto. dnsip si trovava in una custom_components e l'ho rimossa. Tutto ok ora. Saluti e grazie

**Comment by mickyVi** [comment]
Ciao,

io invece continuo a non comprendere bene il funzionamento (tra l'altro mi da sempre errore di compilazione quando provo a fare la modifica), ma piu' che altro perche' ho diviso il configuration.yaml in tanti file che gestiscono le varie entita' o integrazioni (come consigliato in un vostro tutorial su indmus).

Nel configuration.yaml, l'unico command_line che trovo e' quello associato a due sensori per il Rasp:

`## SENSOR
sensor: !include configurations/sensor.yaml`

e al suo interno c'e' questo:

- platform: command_line
  name: Temperatura CPU
  command: "cat /sys/class/thermal/thermal_zone0/temp"
  unit_of_measurement: "°C"
  value_template: '{{ value | multiply(0.001) | round(1) }}'
  scan_interval: 10

Che ho provato a modificare in questo modo:

command_line:
- sensor
    name: Temperatura CPU
    command: "cat /sys/class/thermal/thermal_zone0/temp"
    unit_of_measurement: "°C"
    value_template: '{{ value | multiply(0.001) | round(1) }}'
    scan_interval: 10


Ma ottengo solo un gran bel errore, dove sbaglio e perche' non riesco a capire il mio sbaglio e correggerlo ?

Grazie

ps: queste le modifiche effettuate direttamente nel configuration.yaml, 

![Cattura](https://github.com/indomus/forum/assets/29481285/2a680a96-3c84-47cd-9155-de504d67b316)

L'errore di nuova configurazione e' sparito, pero' l'entita' nno la vede piu:

![Cattura2](https://github.com/indomus/forum/assets/29481285/b91662de-9971-45bf-8f9d-ea73230aaab6)

**Reply by snakuzzo** [reply]
Quello dello screenshot non è un errore...semplicemente l'entità non si chiama `sensor.cpu`. Se vedi bene nel tuo YAML si chiama `sensor.raspcpu`

P.S. Per cortesia correggi il post utilizzando il markdown, altrimenti non si capisce una fava con le indentazioni. Grazie.

**Reply by mickyVi** [reply]
Ciao e intanto grazie,

con il markdown proprio non mi riesce, il - mi diventa sempre un punto elenco.

Proviamo cosi', questo e' il codice originale dentro sensor.yaml
[sensorTest.txt](https://github.com/indomus/forum/files/12162711/sensorTest.txt)

Questo e' quello che ho modificato:
[sensorTestMod.txt](https://github.com/indomus/forum/files/12162734/sensorTestMod.txt)

**Reply by snakuzzo** [reply]
Impara il markdown! [LEGGI!](https://github.com/indomus/forum/discussions/1311)

**Reply by snakuzzo** [reply]
detto ciò ti ho scritto una cosa, l'hai almeno letta ?

Riporto...

> Quello dello screenshot non è un errore...semplicemente l'entità non si chiama sensor.cpu. Se vedi bene nel tuo YAML si chiama sensor.raspcpu

**Reply by mickyVi** [reply]
Si, l'ho letta, ma ho stavo riprovando a sistemare e prima volevo vedere se riuscivo a risolvere da solo...

**Reply by mickyVi** [reply]
Ora meglio:

````
# CONTROLLO VELOCITA VENTOLA RASP 
command_line:
  sensor:
    name: Temperatura CPU
    command: "cat /sys/class/thermal/thermal_zone0/temp"
    unit_of_measurement: "°C"
    value_template: '{{ value | multiply(0.001) | round(1) }}'
    scan_interval: 10

````

Almeno con il markup ci sono riuscito ! :-)

**Reply by mickyVi** [reply]
Ora funziona, ho pero' dei dubbi:

perche' prima della modifica al command_line andava bene come sensor.cpu ora, con la modifica, e' cambiato il nome dell'entita' ?
Che per la cronaca ora e' sensor.temperatura_cpu, e infatti funziona.
Per mantenere lo stesso nome avrei dovuto utilizzare:

`unique_id: sensor.cpu`

E piu' in generale dove prende il nome che lui assegna all'entita' ?

Grazie e sei stato molto gentile a mettermi sulla strada giusta.

**Reply by Scialla** [reply]
come sarebbe "dove prende il nome"?
glielo dai tu!
`name: Temperatura CPU`

da cui deriva l'entità: `sensor.temperatura_cpu`
